test/e2e/gameserverallocation_test.go

```diff
@@ -260,6 +263,542 @@ func TestCreateFleetAndGameServerPlayerCapacityAllocation(t *testing.T) {
 	require.NotEqual(t, gs1.ObjectMeta.Annotations["agones.dev/last-allocated"], gs2.ObjectMeta.Annotations["agones.dev/last-allocated"])
 }
 
+func TestCounterGameServerAllocation(t *testing.T) {
+	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		t.SkipNow()
+	}
+	t.Parallel()
+	ctx := context.Background()
+	client := framework.AgonesClient.AgonesV1()
+
+	flt := defaultFleet(framework.Namespace)
+	flt.Spec.Template.Spec.Counters = map[string]agonesv1.CounterStatus{
+		"games": {
+			Count:    2,
+			Capacity: 10,
+		},
+	}
+
+	flt, err := client.Fleets(framework.Namespace).Create(ctx, flt.DeepCopy(), metav1.CreateOptions{})
+	assert.NoError(t, err)
+	defer client.Fleets(framework.Namespace).Delete(ctx, flt.ObjectMeta.Name, metav1.DeleteOptions{}) // nolint:errcheck
+	framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(flt.Spec.Replicas))
+
+	fleetSelector := metav1.LabelSelector{MatchLabels: map[string]string{agonesv1.FleetNameLabel: flt.ObjectMeta.Name}}
+	stateAllocated := agonesv1.GameServerStateAllocated
+	ready := agonesv1.GameServerStateReady
+	allocated := allocationv1.GameServerAllocationAllocated
+	unallocated := allocationv1.GameServerAllocationUnAllocated
+
+	testCases := map[string]struct {
+		gsa           allocationv1.GameServerAllocation
+		wantGsaErr    bool                                   // For invalid GSA
+		wantAllocated allocationv1.GameServerAllocationState // For a valid GSA: "allocated" if you expect the GSA to succed in allocating a GameServer, "unallocated" if not
+		wantState     agonesv1.GameServerState
+	}{
+		// TODO: These tests assume packed strategy. Should I also test for distributed? Or is that behavior already sufficiently tested elsewhere in this e2e?
+		"Allocate to same GameServer MinAvailable (available capacity)": {
```

Comment:
These tests assume packed strategy. Should I also test for distributed? Or is that behavior already sufficiently tested elsewhere in this e2e?

Reply:
(Something weird is with the formatting on this file??? So a bit hard to read 🤔 I wonder what happened there)

But for just manipulating Counters, i think only testing packed is fine (I'm assuming we test packed/distributed in priority ordering tests in another place).